packages/dialog/src/vaadin-dialog.js

```diff
@@ -39,19 +37,16 @@ registerStyles(
       flex: 1;
     }
 
-    /* prefixing with the element tag to avoid styling confirm-dialog footer part */
     :host([has-title]) [part='header-content'],
-    footer[part='footer'] {
+    [part='footer'] {
       justify-content: flex-end;
     }
 
-    :host(:not([has-title]):not([has-header])) header[part='header'],
-    :host(:not([has-title])) [part='title'] {
-      display: none;
-    }
-
-    :host(:not([has-footer])) footer[part='footer'] {
-      display: none;
+    :host(:not([has-title]):not([has-header])) [part='header'],
+    :host(:not([has-header])) [part='header-content'],
+    :host(:not([has-title])) [part='title'],
+    :host(:not([has-footer])) [part='footer'] {
+      display: none !important;
```

Comment:
Added `!important` to ensure users can't break this by specifying `display` in their custom theme.